### PR TITLE
Addition for survey properties

### PR DESCRIPTION
**Pull request by MatthieuPE**
Summary of Changes:
- Introduced a new module that includes a class for applying survey properties to a stream.
- Added a sample configuration file for a survey, demonstrating how to configure various survey parameters.
- Provided a notebook example that demonstrates the usage of the stream_sim package, covering:
        - Stream generation
        - Application of survey properties
        - DC2 injection for stream simulation
        - Example of stream detection based on the simulated data.

Additionally, modifications were made to the .gitignore file to exclude large or unnecessary files related to survey properties and DC2 data, which are not needed for the repository but are essential for the notebook to function properly. Since these files are required for the example notebook but are excluded from the repository, it may be helpful to provide an external download link for users to access the necessary data files, or anything else.

Let me know if you think some of this should not be on stream_sim package at all.

### CODE REVIEW

File: config/surveys/LSST.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Configuration file for the survey LSST
+name: LSST
```
Comment (psferguson) [inline config/surveys/LSST.yaml:2]: I think this should be DC2 or LSST_DC2 not LSST

File: notebooks/injection_lsst_example.ipynb
Comment (psferguson) [inline notebooks/injection_lsst_example.ipynb:1]: In the "Apply the survey properties" section we mention the files needed, could we also put a link to where we got those files from?

Comment (psferguson) [reply notebooks/injection_lsst_example.ipynb:1]: in "Inject the data into DC2" can we describe what the dp02_dc2_stars.fits file is?
My guess is all objects where extendedness==0, but would be good to document

Comment (psferguson) [reply notebooks/injection_lsst_example.ipynb:1]: cell 17 "plot the results, in the footprint" the density plot seems to have a switched x-axis from later density plots (stream and hotspot are on the left not the right... I think).

Comment (psferguson) [reply notebooks/injection_lsst_example.ipynb:1]: matched filter looks good to me.

File: stream_sim/observed.py [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+        config = self._config.get('survey_files',None)
+        if config is None:
+            raise ValueError('Please specify the files containing informations of the survey.')
+        path = config.get('file_path','')
+        if path=='':
+            current_dir = os.path.dirname(os.path.abspath(__file__)) 
+            path = os.path.join(current_dir, '..', 'data/surveys/'+self.survey)
+        print("Survey's files are searched at the following path : ", path)
+
+        def load_map(attr_name, reader_func, path=path):
```
Comment (psferguson) [inline stream_sim/observed.py:109]: Do we need to have this function nested in another one?

File: notebooks/injection_lsst_example.ipynb
Comment (psferguson) [reply notebooks/injection_lsst_example.ipynb:1]: I'm not sure we want to import generate_stream from bin should we place that function elsewhere?

File: stream_sim/observed.py [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+        flux_g_meas = StreamObserved.magToFlux(mag_g) + np.random.normal(scale=self.getFluxError(mag_g, magerr_g))
+        mag_g_meas = np.where(flux_g_meas > 0., StreamObserved.fluxToMag(flux_g_meas), 99.)
+        flux_r_meas = StreamObserved.magToFlux(mag_r) + np.random.normal(scale=self.getFluxError(mag_r, magerr_r))
+        mag_r_meas = np.where(flux_r_meas > 0., StreamObserved.fluxToMag(flux_r_meas), 99.)
+        
+        return mag_g_meas,mag_r_meas
+
+    def detect_flag(self, pix,mag_r,**kwargs):
+        """
+        Apply the surve selection over the given stars/pixels.
```
Comment (psferguson) [inline stream_sim/observed.py:343]: surve --> survey

File: stream_sim/observed.py [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+        Args:
+            pix (Healpy pixels)
+            mag_r : magnitude in r band
+        Returns:
+            bolean list: 1 for detected stars, 0 for the others
```
Comment (psferguson) [inline stream_sim/observed.py:348]: bolean --> boolean

File: config/surveys/LSST.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    maglim_map_g: supreme_dc2_dr6d_v3_g_maglim_psf_wmean.fits.gz
+    maglim_map_r: supreme_dc2_dr6d_v3_r_maglim_psf_wmean.fits.gz
+    ebv_map: ebv_sfd98_fullres_nside_4096_ring_equatorial.fits
+    completeness: stellar_efficiency_truth.csv
+    log_photo_error: photoerror_r.csv
+    coverage: supreme_dc2_dr6d_v3_gr_fracdet.fits.gz
+survey_properties: 
+  coeff_extinc_g: 3.237
+  coeff_extinc_r: 2.273
+  sys_error : 0.005
```
Comment (psferguson) [inline config/surveys/LSST.yaml:14]: need blank line at end of file

File: stream_sim/observed.py [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+            if extension == 'csv':
+                return pd.read_csv(data)
+            elif extension in ['xls', 'xlsx']:
+                return  pd.read_excel(data)
+            else:
+                raise ValueError(f"Unsupported file format: {extension}")
+        else:
+            raise ValueError(f"Unsupported file format")
+
+    def phi_to_radec(self,phi1,phi2,random=True):
```
Comment (psferguson) [inline stream_sim/observed.py:247]: I think you should also be able to optionally specify endpoints

**Review by psferguson — COMMENTED** [review]
I think my only strong comment is that we should name the survey "LSST_DC2" or "DC2" not LSST (because we should save that name for real data!).

**Comment by kadrlica** [comment]
This is a test comment.

**Review by kadrlica — APPROVED** [review]
Maybe this will make my comments appear?

**Pull request merged**